core/src/main/java/com/segment/analytics/kotlin/core/platform/plugins/logger/SegmentLog.kt

```diff
@@ -0,0 +1,187 @@
+    }
+}
+
+class LogFactory {
+    companion object {
+        fun buildLog(destination: LoggingType.Filter,
+                     title: String,
+                     message: String,
+                     kind: LogFilterKind = LogFilterKind.DEBUG,
+                     function: String? = null,
+                     line: Int? = null,
+                     event: BaseEvent? = null,
+                     sender: Any? = null,
+                     value: Double? = null,
+                     tags: List<String>? = null): LogMessage {
+            return when (destination) {
+                LoggingType.Filter.LOG ->
+                    GenericLog(kind = kind, message = message, function = function, line = line)
+                LoggingType.Filter.METRIC ->
+                    MetricLog(title = title, message = message, value = ((value ?: 1) as Double), event = event, function = function, line = line)
+                LoggingType.Filter.HISTORY ->
+                    HistoryLog(message = message, event = event, function = function, line = line, sender = sender)
+            }
+        }
+    }
+
+    internal class GenericLog(override val kind: LogFilterKind,
+                             override val title: String? = null,
+                             override val message: String,
+                             override val function: String?,
+                             override val line: Int?,
+                             override val event: BaseEvent? = null,
+                             override val logType: LoggingType.Filter = LoggingType.Filter.LOG,
+                             override val dateTime: Date = Date()
+    ): LogMessage
+
+    internal class MetricLog(override val title: String,
+                            override val kind: LogFilterKind = LogFilterKind.DEBUG,
+                            override val message: String,
+                            val value: Double,
+                            override val event: BaseEvent?,
+                            override val function: String?,
+                            override val line: Int?,
+                            override val logType: LoggingType.Filter = LoggingType.Filter.METRIC,
+                            override val dateTime: Date = Date()
+    ): LogMessage
+
+    internal class HistoryLog(override val kind: LogFilterKind = LogFilterKind.DEBUG,
+                             override val title: String? = null,
+                             override val message: String,
+                             override val event: BaseEvent?,
+                             override val function: String?,
+                             override val line: Int?,
+                             val sender: Any?,
+                             override val logType: LoggingType.Filter = LoggingType.Filter.HISTORY,
+                             override val dateTime: Date = Date()
+    ): LogMessage
+}
+
+// Default implementation
+fun LogTarget.flush() {
```

Comment:
this feels like a swift thing but not a kotlin thing. normally u would just make the interface have a default empty implementation (https://github.com/segmentio/analytics-kotlin/pull/56/files#diff-de76397a2e004dcb8d9f3e1537111c83c5adc5148ee82a830dced7538b524efdR27)

Reply:
Ahh that's what i was going for here. I'll refactor.